Fix PotatoSwap adapter by using API when subgraph is unavailable

davidzk3 opened the pull request:
**Summary**
PotatoSwap v2 subgraph is currently unstable due to Cloudflare DNS issues.
This change switches the adapter to use the official PotatoSwap API (rolling 24h)
and falls back to the subgraph only if the API is unavailable.

**Details**
- Uses PotatoSwap v3 API to aggregate v2 pool volume and fees
- Filters strictly to v2 pools
- Removes noisy debug logging
- Avoids retry loops and hanging handles in Node
- Keeps adapter output stable for CI and production

**Notes**
- No changes to testAdapter or shared infra
- Typecheck and adapter tests pass

llamabutler commented:
The potatoswap.ts adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts dexs potatoswap.ts
    
    🦙 Running POTATOSWAP.TS adapter 🦙
    ---------------------------------------------------
    Start Date:	Sun, 28 Dec 2025 00:00:00 GMT
    End Date:	Mon, 29 Dec 2025 00:00:00 GMT
    ---------------------------------------------------
    
    XLAYER 👇
    Backfill start time: 23/4/2024
    Daily volume: 933.28 k
    Daily fees: 2.29 k
    Daily user fees: 2.29 k
    Daily revenue: 0.00
    Daily protocol revenue: 0.00
    Daily supply side revenue: 2.29 k

bheluga reviewed (changes requested):
@davidzk3 Thanks for the PR.
I dont think this is the best solution as we are already tracking potatoswap v3 separately ( https://defillama.com/protocol/potatoswap-v3) , so it would be double counted whenever v2 fails

davidzk3 commented:
> @davidzk3 can you try using https://v3.potatoswap.finance/api/pool/list-all

@bheluga 
Thanks, that makes sense. I’ll switch the fallback to the PotatoSwap API and keep it v2-only. I’ll update the adapter accordingly and push shortly.

llamabutler commented:
The potatoswap.ts adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts dexs potatoswap.ts
    
    🦙 Running POTATOSWAP.TS adapter 🦙
    ---------------------------------------------------
    Start Date:	Tue, 30 Dec 2025 20:32:23 GMT
    End Date:	Wed, 31 Dec 2025 20:32:23 GMT
    ---------------------------------------------------
    
    XLAYER 👇
    Backfill start time: 23/4/2024
    Daily volume: 884.75 k
    Daily fees: 2.21 k
    Daily user fees: 2.21 k
    Daily revenue: 177.00
    Daily protocol revenue: 177.00
    Daily supply side revenue: 2.04 k
    End timestamp: 1767213142 (2025-12-31T20:32:22.000Z)

davidzk3 commented:
@bheluga 
Fixed the PotatoSwap adapter to use the official API when the v2 subgraph is unavailable (Cloudflare DNS issues).
Kept the change adapter-scoped only, removed noisy logs, and ts-check + adapter tests are passing now.
Appreciate a look when you have a moment.
